pkg/proxy/proxy_test.go

```diff
@@ -605,6 +606,50 @@ func (s *TestProxySuite) TestGetWorkspaceContext() {
 			expectedPath:      "/workspace/myworkspace/api",
 			expectedErr:       "",
 		},
+		"valid workspace context with route": {
+			path:              "/plugins/tekton-results/workspaces/myworkspace/api",
+			expectedWorkspace: "myworkspace",
+			expectedPath:      "/api",
+			expectedErr:       "",
+			expectedProxy:     "tekton-results",
+		},
+		// the pre-plugin testcases all assumed /workspace would exists or /api/... exists; should we assume /plugins/<plugin name>/workspace or /plugins/<plugin name/api/... exists?
+		"no specific plugin segment": {
+			path:              "/plugins",
+			expectedWorkspace: "",
+			expectedPath:      "/plugins",
+			expectedErr:       "path \"/plugins\" not a proxied route request",
+			expectedProxy:     "",
+		},
+		// the pre-plugin testcases all assumed /workspace would exists or /api/... exists; should we assume /plugins/<plugin name>/workspace or /plugins/<plugin name/api/... exists?
+		"plugin spec but nothing else": {
+			path:              "/plugins/whatever",
+			expectedWorkspace: "",
+			expectedPath:      "",
+			expectedErr:       "",
+			expectedProxy:     "whatever",
+		},
+		"invalid workspace context with route": {
+			path:              "/plugins/tekton-results/workspaces/myworkspace",
+			expectedWorkspace: "",
+			expectedPath:      "/workspaces/myworkspace",
+			expectedErr:       "workspace request path has too few segments '/workspaces/myworkspace'; expected path format: /workspaces/<workspace_name>/api/...",
+			expectedProxy:     "tekton-results",
+		},
+		"no workspace context with route": {
+			path:              "/plugins/tekton-results/api/pods",
+			expectedWorkspace: "",
+			expectedPath:      "/api/pods",
+			expectedErr:       "",
+			expectedProxy:     "tekton-results",
+		},
+		"workspace instead of workspaces with route": {
+			path:              "/plugins/tekton-results/workspace/myworkspace/api",
+			expectedWorkspace: "",
+			expectedPath:      "/workspace/myworkspace/api",
+			expectedErr:       "",
+			expectedProxy:     "tekton-results",
+		},
```

Comment:
Let's add two more test cases which would test that the plugin name is expected in the beginning of the path only:
- `/plugins/tekton-results/workspaces/myworkspace/api/plugins/something` should work. In this case the actual kube api contains "plugins" which has nothing to do with our proxy plugins.
- `/api/plugins/something`. There is no plugins or workspace context. But the kube api still includes some "plugins". Still should work.